modules/nf-core/ensemblvep/main.nf

```diff
@@ -32,16 +29,17 @@ process ENSEMBLVEP {
     script:
     def args = task.ext.args ?: ''
     def file_extension = args.contains("--vcf") ? 'vcf' : args.contains("--json")? 'json' : args.contains("--tab")? 'tab' : 'vcf'
-    def compress_out = args.contains("--compress_output") ? '.gz' : ''
+    def compress_cmd = args.contains("--compress_output") ? '' : '--compress_output bgzip'
     def prefix = task.ext.prefix ?: "${meta.id}"
     def dir_cache = cache ? "\${PWD}/${cache}" : "/.vep"
     def reference = fasta ? "--fasta $fasta" : ""
 
     """
     vep \\
         -i $vcf \\
-        -o ${prefix}.ann.${file_extension}${compress_out} \\
+        -o ${prefix}.ann.${file_extension}.gz \\
```

Comment:
then prefix is always with `.gz`?

Reply:
Not `$prefix`, but the output, yes. It'll be compressed regardless of the users args and have a `.gz` extension.
Choices within vep are bgzip or gzip, so `.gz` either way